### ISSUE TITLE
hostname conflict with avahi (post 3.0)

### ISSUE THREAD

**Issue by Stibbons** [pi-hole/pi-hole#1426]
**In raising this issue, I confirm the following (please check boxes, eg [X]) Failure to fill the template will close your issue:**

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*
- [x] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the codebase?:**

4

---
**[BUG] Expected Behaviour:**
Should be able to ssh to the pi-hole raspberry using "hostname.local". "local" suffix is given by my DSL box's DHCP.

**[BUG] Actual Behaviour:**
After a few seconds upon start up, DNS resolution is not possible, only possible to access through the static IP.

Logs shows Avahi conflicts with unexisting hosts:
```
May  3 08:14:14 pihole daemon.info avahi-daemon[346]: Registering new address record for 2a01:e35:2ecf:6500:18cb:2317:acdc:c685 on eth0.*.
May  3 08:14:14 pihole daemon.info avahi-daemon[346]: Registering new address record for 192.168.0.3 on eth0.IPv4.
May  3 08:14:14 pihole daemon.info avahi-daemon[346]: Registering HINFO record with values 'ARMV7L'/'LINUX'.
May  3 08:14:34 pihole daemon.info avahi-daemon[346]: Withdrawing address record for 2a01:e35:2ecf:6500:18cb:2317:acdc:c685 on eth0.
May  3 08:14:34 pihole daemon.warn avahi-daemon[346]: Host name conflict, retrying with pihole-1201
May  3 08:14:34 pihole daemon.info avahi-daemon[346]: Registering new address record for 2a01:e35:2ecf:6500:18cb:2317:acdc:c685 on eth0.*.
May  3 08:14:34 pihole daemon.info avahi-daemon[346]: Registering new address record for 192.168.0.3 on eth0.IPv4.
May  3 08:14:34 pihole daemon.info avahi-daemon[346]: Registering HINFO record with values 'ARMV7L'/'LINUX'.
May  3 08:14:54 pihole daemon.info avahi-daemon[346]: Withdrawing address record for 2a01:e35:2ecf:6500:18cb:2317:acdc:c685 on eth0.
May  3 08:14:54 pihole daemon.warn avahi-daemon[346]: Host name conflict, retrying with pihole-1202
May  3 08:14:54 pihole daemon.info avahi-daemon[346]: Registering new address record for 2a01:e35:2ecf:6500:18cb:2317:acdc:c685 on eth0.*.
May  3 08:14:54 pihole daemon.info avahi-daemon[346]: Registering new address record for 192.168.0.3 on eth0.IPv4.
May  3 08:14:54 pihole daemon.info avahi-daemon[346]: Registering HINFO record with values 'ARMV7L'/'LINUX'.
```
**[BUG] Workaround:**

I [found a workaround](https://bugs.launchpad.net/ubuntu/+source/avahi/+bug/831022):
- remove the last line of `/etc/hosts`, the one with ```127.0.1.1  pihole```

After this, on reboot no more issue. Web interface and ssh back to normal

**Comment by WaLLy3K** [comment]
I believe this issue may be resolved by #1408

**Comment by Stibbons** [comment]
Confirmed. This fixes this issue (so no need to edit `/etc/hosts`). Great !

**Comment by PromoFaux** [comment]
Fixed in #1408

**Comment by Stibbons** [comment]
Sadly it happens again, it actually took a few dozen of minutes after restart to fail again.

Fixed by removing the line of `/etc/hosts`.
